client/state/sites/sharing-buttons/actions.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SHARING_BUTTONS_RECEIVE,
+	SHARING_BUTTONS_REQUEST,
+	SHARING_BUTTONS_REQUEST_FAILURE,
+	SHARING_BUTTONS_REQUEST_SUCCESS,
+	SHARING_BUTTONS_SAVE,
+	SHARING_BUTTONS_SAVE_FAILURE,
+	SHARING_BUTTONS_SAVE_SUCCESS,
+	SHARING_BUTTONS_UPDATE
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that sharing buttons have been received.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The sharing buttons object
+ * @return {Object}        Action object
+ */
+export function receiveSharingButtons( siteId, settings ) {
+	return {
+		type: SHARING_BUTTONS_RECEIVE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that some sharing buttons have been updated.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The updated sharing buttons
+ * @return {Object}        Action object
+ */
+export function updateSharingButtons( siteId, settings ) {
+	return {
+		type: SHARING_BUTTONS_UPDATE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action thunk which, when invoked, triggers a network request to
+ * retrieve sharing buttons
+ *
+ * @param  {Number} siteId Site ID
+ * @return {Function}      Action thunk
+ */
+export function requestSharingButtons( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SHARING_BUTTONS_REQUEST,
+			siteId
+		} );
+
+		return wpcom.undocumented().sharingButtons( siteId )
```

Comment:
To be honest, I haven't in this PR. How would you  handle the "is already requesting skip" on the API middleware? would you use, `getState` with a selector or is there any mechanism I'm not aware of? How is this different from using `getState` on the action creator which I think is considered as a bad practice.